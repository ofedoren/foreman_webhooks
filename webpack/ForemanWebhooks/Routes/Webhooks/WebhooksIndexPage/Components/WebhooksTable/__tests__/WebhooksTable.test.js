import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import {
  propsFactory,
  webhooks,
} from '../../../__tests__/WebhooksIndexPage.fixtures';
import WrappedWebhooksTable from '../index';

const props = {
  fetchAndPush: () => {},
  onDeleteClick: () => {},
  setToDelete: () => {},
  itemCount: 0,
  canCreate: true,
};

const fixtures = {
  'should render when loading': propsFactory({
    ...props,
    isLoading: true,
    hasData: false,
    hasError: false,
    toasts: [],
  }),
  'should render with no data': propsFactory({
    ...props,
    isLoading: false,
    hasData: false,
    hasError: false,
    toasts: [],
  }),
  'should render with error': propsFactory({
    isLoading: false,
    hasData: false,
    hasError: true,
    message: {
      type: 'error',
      text: 'this is error',
    },
    ...props,
    toasts: [],
  }),
  'should render with webhooks': propsFactory({
    ...props,
    isLoading: false,
    hasError: false,
    hasData: true,
    toasts: [],
    webhooks,
    itemCount: webhooks.length,
  }),
};

describe('WebhooksTable', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(WrappedWebhooksTable, fixtures));
});
