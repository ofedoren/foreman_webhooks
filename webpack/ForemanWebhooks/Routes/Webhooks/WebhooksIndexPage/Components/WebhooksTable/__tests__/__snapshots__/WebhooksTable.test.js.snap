// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebhooksTable rendering should render when loading 1`] = `
<Component
  canCreate={true}
  fetchAndPush={[Function]}
  hasData={false}
  hasError={false}
  isLoading={true}
  itemCount={0}
  onDeleteClick={[Function]}
  page={5}
  perPage={42}
  search="name=foo"
  sort={
    Object {
      "by": "defaultName",
      "order": "ASC",
    }
  }
  toasts={Array []}
/>
`;

exports[`WebhooksTable rendering should render with error 1`] = `
<Component
  canCreate={true}
  fetchAndPush={[Function]}
  hasData={false}
  hasError={true}
  isLoading={false}
  itemCount={0}
  message={
    Object {
      "text": "this is error",
      "type": "error",
    }
  }
  onDeleteClick={[Function]}
  page={5}
  perPage={42}
  search="name=foo"
  sort={
    Object {
      "by": "defaultName",
      "order": "ASC",
    }
  }
  toasts={Array []}
/>
`;

exports[`WebhooksTable rendering should render with no data 1`] = `
<Component
  canCreate={true}
  fetchAndPush={[Function]}
  hasData={false}
  hasError={false}
  isLoading={false}
  itemCount={0}
  onDeleteClick={[Function]}
  page={5}
  perPage={42}
  search="name=foo"
  sort={
    Object {
      "by": "defaultName",
      "order": "ASC",
    }
  }
  toasts={Array []}
/>
`;

exports[`WebhooksTable rendering should render with webhooks 1`] = `
<Component
  canCreate={true}
  fetchAndPush={[Function]}
  hasData={true}
  hasError={false}
  isLoading={false}
  itemCount={2}
  onDeleteClick={[Function]}
  page={5}
  perPage={42}
  search="name=foo"
  sort={
    Object {
      "by": "defaultName",
      "order": "ASC",
    }
  }
  toasts={Array []}
  webhooks={
    Array [
      Object {
        "canDelete": true,
        "canEdit": true,
        "enabled": true,
        "id": 1,
        "name": "my-webhook",
        "targetUrl": "https://my-machine.example.com",
      },
      Object {
        "canDelete": false,
        "canEdit": false,
        "enabled": false,
        "id": 2,
        "name": "your-webhook",
        "targetUrl": "https://your-machine.example.com",
      },
    ]
  }
/>
`;
